client/src/views/Meme.svelte

```diff
@@ -1 +1,150 @@
-<h1>Single Meme</h1>
+<script>
+	import Comment from "../lib/View/Comment.svelte";
+	import { memes } from "../cache";
+	export let params = {};
+	let image;
+	let upvotes;
+	let views;
+	let comments;
+	let title;
+	let myComment;
+	let description;
+	let id = params.id;
+	let meme = $memes.get(id);
+
+	update();
+
+
+	function submit(){
+		comments.push({ name:"Ines",text:myComment });
+		comments = comments; //für reaktivität von svelte
+		myComment = "";
+
+	}
+	function nextImage(){
+		if(id < $memes.size - 1){
+			id = String(parseInt(id) + 1);
+			meme = $memes.get(id);
+		}
+		else{
+			id= '0';
+			meme = $memes.get('0');
+		}
+		update();
+	}
+	function lastImage(){
+		if(id > 0){
+			id = String(parseInt(id) - 1);
+			meme = $memes.get(id);
+		}
+		else{
+			id = String($memes.size-1);
+			meme = $memes.get(id);
+		}
+		update();
+
+	}
+	function update(){
```

Comment:
Also diese ganze Function hier... Im Prinzip der Fakt dass du von all diesen fields von `Meme` hier Kopien hast ist doch eigentlich gar nicht nötig oder? Also damit machst du's dir glaube nur schwieriger und das Prinzip der reaktivität so ein bisschen kaputt.

Spricht was dagegen, dass du einfach direkt immer `$meme` verwendest? Also wo auch immer du z. B. deine lokale Variable `comments` brauchst... Warum nicht einfach direkt `$meme.comments` nehmen und modifizieren, bzw anzeigen? Dann brauchst du nicht dieses update-gedöns hier.

***

Generell aber als Tipp, wenn du mal wirklich was updaten musst auf Basis von _anderen_ reaktiven Variablen. Da hat Svelte diese magische Syntax wieder mit `$`. Siehe https://svelte.dev/tutorial/reactive-declarations

Also z. B.
```js
$: comments = $meme.comments;
```

...würde tatsächlich dafür sorgen, dass `comments` immer dem Wert vom `$meme.comments` entspricht. Aber hier in deinem Fall ist das glaube eher ein Antipattern und man sollte einfach direkt `$meme.comments` verwenden ohne lokale Kopie.

Reply:
Hab ich geändert